### PR TITLE
Enable and fix new lints

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: _test_yaml/test/ensure_build_test.dart [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 @TestOn('vm')
 @Tags(['presubmit-only'])
+library test;
```
Comment (kevmoo) [inline _test_yaml/test/ensure_build_test.dart:7]: To be cleaned up when we have https://github.com/dart-lang/linter/issues/4031

Comment (jakemac53) [inline _test_yaml/test/ensure_build_test.dart:7]: what is up with these?

Comment (kevmoo) [reply _test_yaml/test/ensure_build_test.dart:7]: We don't want dangling annotations. These are annotating the LIBRARY.

Comment (jakemac53) [reply _test_yaml/test/ensure_build_test.dart:7]: oh I see, it does make things more stable I guess so sgtm

**Review by jakemac53 — APPROVED** [review]

**Pull request merged**